main.go

```diff
@@ -218,3 +249,85 @@ func addPodSecurityPrivilegedLabels(watchNamespaceName string) error {
 	}
 	return nil
 }
+
+func DoesCRDExist(CRDName string) (bool, error) {
+	kubeconf := ctrl.GetConfigOrDie()
+
+	dynamicClient, err := dynamic.NewForConfig(kubeconf)
+
+	if err != nil {
+		return false, err
+	}
+
+	gvr := schema.GroupVersionResource{
+		Group:    "apiextensions.k8s.io",
+		Version:  "v1",
+		Resource: "customresourcedefinitions",
+	}
+
+	crd, err := dynamicClient.Resource(gvr).Get(context.Background(), CRDName, metav1.GetOptions{})
+	if err != nil {
+		setupLog.Error(err, "error checking for CRDs existence")
+		return false, err
+	}
+
+	// Check if the CRD is found and has a non-empty UID
+	if crd != nil && crd.GetUID() != "" {
+		setupLog.Info(fmt.Sprintf("crd exists with UID: %s", crd.GetUID()))
+		return true, nil
+	}
+
+	return false, nil
+
+}
+
+// CreateCredRequest WITP : WebIdentityTokenPath
+func CreateCredRequest(roleARN string, WITP string, secretNS string) error {
+	cfg := config.GetConfigOrDie()
+	client, err := client2.New(cfg, client2.Options{})
+	if err != nil {
+		setupLog.Error(err, "unable to create client")
+	}
+
+	credRequest := &unstructured.Unstructured{
```

Comment:
could we import it from here https://pkg.go.dev/github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1#CredentialsRequest or adds to much size to OADP?

Reply:
So initially that is what I did but that added some additional updates to go.mod and started resulting in build failures (incompatible versions with velero deps), hence going via the unstructured route. Context comment: https://github.com/openshift/oadp-operator/pull/1106#issuecomment-1678186905

Reply:
alright, thanks for the info!

Reply:
Done, added a comment.